### ISSUE TITLE
Occasional FPEs (div by zero) when AI tries to throw a grenade

### ISSUE THREAD

**Issue by oldlaptop** [ja2-stracciatella/ja2-stracciatella#1006]
I've noticed in the last few months (verified on the indicated hash) that the game occasionally bombs with a floating-point exception; upon further investigation it's a division by zero at https://github.com/ja2-stracciatella/ja2-stracciatella/blob/master/src/game/TacticalAI/Attacks.cc#L944. Digging to the bottom of it is unfortunately more time than I have to devote to this (not nearly familiar enough with the codebase). I'm afraid I don't have a reliable repro, but the attached save "often" crashes: move Carlos one square to the "west", so as to get LoS (once the smoke clears) on the row of squares between the burned out tank and the sandbags, and the AI will *usually* shoot him (once the smoke clears) and then trigger the crash. Github identifies #701 as possibly related, but I'm not able to evaluate that possibility.

I don't recall ever seeing a crash in this sort of situation in vanilla or in older (before the past 6-8 months or so) Stracciatella, but of course I can't be certain. It does seem to occur often enough that I think I'd have noticed earlier if it was present.

Tested on voidlinux, gcc 9.2.0; other environment attributes available on request. 

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
| 38f5f8f3ddaf21d29c224c33a7f9c856150e3f6f  | JA2 from GOG  |    unknown, suspected no    |    unknown, suspected yes     |

[fpe.zip (happens to be a combat quicksave)](https://github.com/ja2-stracciatella/ja2-stracciatella/files/4259526/fpe.zip)

**Comment by flaviojs** [comment]
Looking at that line of code:
https://github.com/ja2-stracciatella/ja2-stracciatella/blob/38f5f8f3ddaf21d29c224c33a7f9c856150e3f6f/src/game/TacticalAI/Attacks.cc#L944
`ubRawAPCost` and `ubMaxPossibleAimTime` must be 0 to get a divide by zero.

`ubMaxPossibleAimTime` is built from the remaining action points so it can be 0 at any time. 
After trying the save I found that `ubRawAPCost` is 0 for `GLAUNCHER`:
```
2020-03-02T22:03:00 [ WARN] game/Tactical/Points.cc: MinAPsToThrow - in-hand item 'GLAUNCHER' has unexpected item class 0x10
```

This behavior and warning was introduced in #287, where an ancient check in MinAPsToThrow was fixed.

---

Now for the real bug...

In `MinAPsToAttack` the item class `IC_LAUNCHER` (0x10) calls `MinAPsToShootOrStab` but this ancient comment implies it should be using `MinAPsToThrow`:

https://github.com/ja2-stracciatella/ja2-stracciatella/blob/38f5f8f3ddaf21d29c224c33a7f9c856150e3f6f/src/game/TacticalAI/Attacks.cc#L924

What needs fixing? Is it `MinAPsToAttack` or the comment and this code?

Or in other words, how should we classify the use of a grenade launcher? A shoot/stab or a throw or something else? (affects AP and chance to hit in this function)

**Comment by lynxlynxlynx** [comment]
Might be easiest to just look at the ap cost in the original and compare it with the two options.

**Comment by flaviojs** [comment]
A quick fix for the FPE is to not return 0 inside the fixed check in `MinAPsToThrow` and just continue, it would produce the same APs as before the ancient check was fixed.

However, the inconsistency in behavior for `GLAUNCHER` that I found is likely to repeat in other places of the spaghetti code, so the correct behavior should be properly defined. I don't know enough of the game and weapons to judge what is the correct behavior.
